src/expression.js

```diff
@@ -206,7 +206,7 @@ pp.parseExprSubscripts = function(refShorthandDefaultPos) {
   let startPos = this.start, startLoc = this.startLoc
   let expr = this.parseExprAtom(refShorthandDefaultPos)
   if (refShorthandDefaultPos && refShorthandDefaultPos.start) return expr
-  return this.parseSubscripts(expr, startPos, startLoc)
+  return this.parseSubscripts(expr, startPos, startLoc, expr.type === "ArrowFunctionExpression" && this.parenthesesCount === 0)
```

Comment:
I tried that first but couldn't get it to work. The problem is that the opening `(` of the arrow function triggers `p_expr`, so everytime `expr.type` is "ArrowFunctionExpression" where the args are in parens, the current context always ends up as `p_expr`.